addons/multiplegap/src/presenter.js

```diff
@@ -1594,17 +1594,49 @@ function Addonmultiplegap_create(){
     MultipleGapKeyboardController.prototype.enter = function (event) {
         window.KeyboardController.prototype.enter.call(this, event);
         if(this.keyboardNavigationElementsLen > 1) {
-            var voicesArray = [];
-            for(var i  = 1; i <this.keyboardNavigationElementsLen; i++) {
-                var $element = this.getTarget(this.keyboardNavigationElements[i]);
-                voicesArray = voicesArray.concat(presenter.getTextVoicesFromPlaceholder($element));
+            if (presenter.configuration.sourceType === presenter.SOURCE_TYPES.AUDIO) {
+                var $element = presenter.$view.find('.placeholder.keyboard_navigation_active_element');
+                if ($element.length == 0) {
+                    readAllElements();
+                } else {
+                    var draggableItemValues = $element.attr('draggableitem').split('-');
+                    var $wrapper = $element.find('.multiaudio-item-wrapper');
+                    var callback = function() {
+                        if (draggableItemValues.length == 2) {
+                            playDraggableAudio($wrapper, draggableItemValues[1], draggableItemValues[0]);
+                        }
+                    }
+                    var voicesArray = [];
+                    voicesArray = voicesArray.concat(presenter.getTextVoicesFromPlaceholder($element));
```

Comment:
Chyba nie ma potrzeby tutaj concatować pustej tablicy ze zwrotką z getTextVoicesFromPlaceholder?
A = [];
B = [1,2,3]
A.concat(B) to zawsze będzie zawartość B (a getTextVoicesFromPlaceholder zawsze tworzy nową tablicę, więc nie ma obawy o modyfikacje jakiegos elementu)